core/object.go

```diff
@@ -79,6 +79,11 @@ func (t ObjectType) Bytes() []byte {
 	return []byte(t.String())
 }
 
+// Valid returns true if t is a valid ObjectType.
+func (t ObjectType) Valid() bool {
```

Comment:
I meant to use this to check the validity of the provided object type in objfile.NewWriter. I'll update the PR.

Object types default to 0, which is an invalid value, so I added this function for a consistent way to check that someone didn't forget to provide a valid value.